webview-ui/test-generation/src/App.tsx

```diff
@@ -115,26 +122,45 @@ function App() {
     status,
     steps,
     tunnel,
+    apps,
   } = state;
 
   return (
     <main>
       <section className="inputs">
         <h2>What do you want to test?</h2>
-        <VSCodeTextField
-          value={appName}
-          placeholder="From Sauce Labs App Storage, e.g. test.apk"
-          onInput={(e) => {
-            if (e.target && 'value' in e.target) {
-              dispatch({
-                type: 'setAppName',
-                value: e.target.value as string,
-              });
-            }
-          }}
-        >
-          Application Name
-        </VSCodeTextField>
+        <section className="with-label">
+          <label>App Name</label>
+          <VSCodeDropdown
+            onInput={(e) => {
+              if (e.target && 'value' in e.target) {
+                const appName: string = e.target.value as string;
+                const appInfo = apps.find(
+                  (app) => app.name === appName,
+                ) as AppInfo;
+                dispatch({
+                  type: 'setAppName',
+                  value: {
+                    appName: appName as string,
```

Comment:
https://github.com/saucelabs/vscode-scriptiq/blob/7b7798be8bf82ec139a3453a1fb8ac0423c24560/webview-ui/test-generation/src/App.tsx#L137-L137 is already doing a cast.